helix-rest/src/main/java/org/apache/helix/rest/server/ServerContext.java

```diff
@@ -117,9 +121,20 @@ public HelixDataAccessor getDataAccssor(String clusterName) {
     }
   }
 
+  public ZkBaseDataAccessor<ZNRecord> getZkBaseDataAccessor(ZkSerializer serializer) {
+    synchronized (_zkBaseDataAccessorBySerializer) {
+      if (!_zkBaseDataAccessorBySerializer.containsKey(serializer)) {
```

Comment:
Did you mean to do double-lock checking here? Like I pointed out in a different PR, we need to follow the format given here:

https://stackoverflow.com/questions/3578604/how-to-solve-the-double-checked-locking-is-broken-declaration-in-java

If you synchronize at the beginning, it will not allow parallel access to the map, which affect parallelism and therefore performance.

Reply:
Fixed. It's merely the existing implementations I copied from other methods in the same class.